### PR TITLE
Compile a single entry into multiple files

### PR DESCRIPTION
**Pull request by axelboc**
This is done by returning an array of paths from the custom `path` function. The main use case is i18n (cf. updated README).

### CODE REVIEW

**Comment by kylemac** [comment]
@axelboc sorry we've been slow with this one. reviewing now and trying to get this merged for you. i'll leave some comments if I find them

File: lib/index.coffee [hunk 1]
```diff
@@ -118,7 +118,8 @@ module.exports = (opts) ->
       _.assign(_.omit(e, 'fields'), e.fields)
 
     ###*
-     * Sets `_url` property on content with single entry views
+     * Sets `_url` and `_urls` properties on content with single entry views
+     * `_url` takes the value `null` if the content type's custom path function returns multiple paths
```
Comment (kylemac) [inline lib/index.coffee]: can you drop to a new line after `path function` so we keep it within our 80 char limit?

File: lib/index.coffee [hunk 2]
```diff
@@ -148,11 +152,15 @@ module.exports = (opts) ->
...
-          @roots.config.locals.entry = entry
           compiler = _.find @roots.config.compilers, (c) ->
             _.contains(c.extensions, path.extname(template).substring(1))
-          compiler.renderFile(template, @roots.config.locals)
-            .then((res) => @util.write("#{t.path(entry)}.html", res.result))
+          W.map entry._urls, (url) =>
+            @roots.config.locals.entry = _.assign({}, entry, { _url: url })
+            compiler.renderFile(template, @roots.config.locals)
+              .then((res) =>
+                @roots.config.locals.entry = null
```
Comment (hhsnopek) [inline lib/index.coffee]: why did you have to set this to null even though you set the entry above?

File: lib/index.coffee [hunk 3]
```diff
@@ -126,7 +127,10 @@ module.exports = (opts) ->
     set_urls = (types) ->
       W.map types, (t) ->
         if t.template then W.map t.content, (entry) ->
-          entry._url = "/#{t.path(entry)}.html"
+          paths = t.path(entry)
+          paths = [paths] if typeof paths == 'string'
```
Comment (hhsnopek) [inline lib/index.coffee]: Change this to `paths = [paths ] if _.isString(paths)`

File: lib/index.coffee [hunk 4]
```diff
@@ -126,7 +127,10 @@ module.exports = (opts) ->
...
+          entry._urls = ("/#{p}.html" for p in paths)
+          entry._url = if entry._urls.length == 1 then entry._urls[0] else null
```
Comment (hhsnopek) [inline lib/index.coffee]: Could you change `==` to `is`

**Comment by hhsnopek** [comment]
see #36

**Comment by axelboc** [comment]
Thanks! Sorry I haven't been able to get to back into this.

**Comment by hhsnopek** [comment]
@axelboc out of curiosity, with your readme section, how are you applying i18n changes with a single template or one locale?

**Comment by axelboc** [comment]
In Contentful, I prefix each localised field with `fr` or `en`. In the entry template, I parse the URL (`entry._url`) in order to get the language code, and use it to access the fields - e.g. `entry[lang + 'Title']`. It's not as convenient as locales, but I can't afford Contenful's cheapest paid plan (by far)...

**Comment by hhsnopek** [comment]
Would you mind doing a little write up on that for others and add it to your PR?

**Comment by axelboc** [comment]
In the README? Sure! :) Won't be for a couple of weeks, though.

**Comment by hhsnopek** [comment]
yeah, thank you! It'll be valuable for others to know - I'll get the branch merged in then, considering the timeline on that :)

**Comment by axelboc** [comment]
Okay, thanks!

**Comment by hhsnopek** [comment]
now that I think about it, you should throw it in http://roots.cx ([roots-www](https://github.com/carrot/roots-www)) as an article!

**Pull request closed without merging**